x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/field_parameters/fielddata_parameter.tsx

```diff
@@ -4,34 +4,96 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { EuiSpacer, EuiFormRow, EuiDualRange, EuiCallOut, EuiLink } from '@elastic/eui';
+import { EuiSpacer, EuiFormRow, EuiCallOut, EuiLink } from '@elastic/eui';
 
 import {
   UseField,
   Field,
   FormDataProvider,
-  FieldHook,
   UseMultiFields,
+  FieldHook,
 } from '../../../shared_imports';
 import { getFieldConfig } from '../../../lib';
+import { NormalizedField } from '../../../types';
 import { EditFieldFormRow } from '../fields/edit_field';
 import { documentationService } from '../../../../../services/documentation';
+import { FielddataFrequencyFilterPercentage } from './fielddata_frequency_filter_percentage';
+import { FielddataFrequencyFilterAbsolute } from './fielddata_frequency_filter_absolute';
 
 interface Props {
   defaultToggleValue: boolean;
+  field: NormalizedField;
 }
 
-export const FieldDataParameter = ({ defaultToggleValue }: Props) => {
-  const onFrequencyFilterChange = (minField: FieldHook, maxField: FieldHook) => ([
-    min,
-    max,
-  ]: any) => {
-    minField.setValue(min);
-    maxField.setValue(max);
+type ValueType = 'percentage' | 'absolute';
+
+export const FieldDataParameter = ({ field, defaultToggleValue }: Props) => {
+  const [valueType, setValueType] = useState<ValueType>(
+    field.source.fielddata_frequency_filter !== undefined
+      ? (field.source.fielddata_frequency_filter as any).max > 1
+        ? 'absolute'
+        : 'percentage'
+      : 'percentage'
+  );
+
+  const getConfig = (fieldProp: 'min' | 'max', type = valueType) =>
+    type === 'percentage'
+      ? getFieldConfig('fielddata_frequency_filter_percentage', fieldProp)
+      : getFieldConfig('fielddata_frequency_filter_absolute', fieldProp);
+
+  const switchType = (min: FieldHook, max: FieldHook) => () => {
+    const nextValueType = valueType === 'percentage' ? 'absolute' : 'percentage';
+    const nextMinConfig = getConfig('min', nextValueType);
+    const nextMaxConfig = getConfig('max', nextValueType);
+
+    min.setValue(
+      nextMinConfig.deserializer
+        ? nextMinConfig.deserializer(nextMinConfig.defaultValue)
+        : nextMinConfig.defaultValue
```

Comment:
Good idea, at first sight, it seems more difficult to understand, but I guess we need to train our brain for it as it will be everywhere in 6 months! 😄